### PR TITLE
[Bndtools Core] Fixed Content Assist

### LINKED ISSUE #2974: Several Instructions and Headers ain't displayed in Content Assist
In Bnd Source View, we can use CTRL + SPACE to trigger the content assist but lots of instructions and headers ain't displayed even though they do exist. For example, `-nobundles`, `-workingset`, `-privatepackages`. 

Related Bug: #2937

### PR DESCRIPTION
**Pull request by amitjoy**
Content Assist couldn't list all existent instructions and headers. This fix solves this problem.

Fixes #2974 

Signed-off-by: Amit Kumar Mondal <admin@amitinside.com>

### CODE REVIEW

File: bndtools.core/src/bndtools/editor/completion/BndCompletionProcessor.java [hunk 1]
```diff
@@ -45,15 +50,13 @@
...
-                    : s.getHeader()
-                        .substring(prefix.length());
-                results.add(new CompletionProposal(text + ": ", offset, 0, text.length() + 2, null, s.getHeader(), info, s.getLead())); //$NON-NLS-1$
+        List<ICompletionProposal> results = new ArrayList<>();
+
+        for (String s : ALL_OPTIONS) {
+            if (prefix == null || s.startsWith(prefix)) {
+                IContextInformation info = new ContextInformation(s, s);
+                String text = prefix == null ? s : s.substring(prefix.length());
+                results.add(new CompletionProposal(text + ": ", offset, 0, text.length() + 2, null, s, info, null));
```
Comment (rotty3000) [inline bndtools.core/src/bndtools/editor/completion/BndCompletionProcessor.java]: I'm not sure it's desired to switch the lookup to not use Syntax.HELP.

Why not just add the missing properties to the HELP, along with some helpful information? You can draw that form the existing docs.

Comment (amitjoy) [reply bndtools.core/src/bndtools/editor/completion/BndCompletionProcessor.java]: The following list comprises the instructions that are missing in `Syntax`.

```
CONSUMER_POLICY
NOBUNDLES
PROVIDER_POLICY
RUNSYSTEMCAPABILITIES
REPORTNEWER
UNDERTEST
TESTPACKAGES
NOMANIFEST
SAVEMANIFEST
WAB
WABLIB
RUNFRAMEWORK
RUNFW
RUNKEEP
RUNTRACE
TESTCONTINUOUS
SNAPSHOT
NAMESECTION
DIGESTS
DSANNOTATIONS
DSANNOTATIONS_OPTIONS
BASELINEREPO
EXECUTABLE
RUNNOREFERENCES
JAVAAGENT
STRICT
DIFFIGNORE
DIFFPACKAGES
CONTRACT
NOBUILDINCACHE
EXTENSION
NOJUNIT
NOJUNITOSGI
PREPROCESSMATCHERS
OUTPUTMASK
TESTUNRESOLVED
RUNJDB
RUNENV
EEPROFILE
EXPORT
GESTALT
BNDDRIVER
CHECK
DISTRO
METATYPE_ANNOTATIONS
METATYPE_ANNOTATIONS_OPTIONS
PACKAGEINFOTYPE
JAVAC_SOURCE
JAVAC_TARGET
JAVAC_PROFILE
JAVAC
JAVA
JAVA_DEBUG
EXPORTTYPE
RUNREMOTE
AUGMENT
REQUIRE_BND
GROUPID
IGNORE_STANDALONE
RUNREPOS
INIT
MAVEN_RELEASE
BUILDREPO
CONNECTION_SETTINGS
RUNPROVIDEDCAPABILITIES
WORKINGSET
RUNSTORAGE
INCLUDEPACKAGE
CDIANNOTATIONS
REMOTEWORKSPACE
SIGNATURE_TEST
```

**Comment by amitjoy** [comment]
@rotty3000 @bjhargrave I tried to add all the remaining headers and instructions to Syntax but still couldn't find any detailed info about the following ones:

```
Constants.IGNORE_STANDALONE           :     -ignore-standalone
Constants.SIGNATURE_TEST              :     -signaturetest
```

I have checked all the references but couldn't find any reference on how it is processed by bnd. There exists only one reference where it is only consumed - `BndEditModel`.

File: biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java [hunk 2]
```diff
@@ -203,18 +220,27 @@
...
 			+ " works as private package but will only include the packages when they are imported. When this header is used, bnd will recursively add packages that match the patterns until there are no more additions.",
 			CONDITIONALPACKAGE + ": com.*", "${packages}", null),
+		new Syntax(META_PERSISTENCE,
+			"A bundle is regarded as a persistence bundle if it contains the header " + META_PERSISTENCE
+				+ " in it's Manifest.",
+			META_PERSISTENCE + ": com.*", "${packages}", null),
 		new Syntax(PRIVATE_PACKAGE, "The " + PRIVATE_PACKAGE
 			+ " header contains a declaration of packages to be included in the resulting bundle, the only difference is, is that these packages will not be exported.",
-			PRIVATE_PACKAGE + ": com.*", "${packages}", null),
+			PRIVATE_PACKAGE + ": persistence/myPu.xml", null, null),
```
[the remark this answers is not in the document]
Comment (amitjoy) [reply biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java]: Ah you are right. Copy-Paste errors 😄

File: biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java [hunk 3]
```diff
@@ -287,15 +313,65 @@
...
+			null, null),
 		new Syntax(BUILDERIGNORE,
 			"List of project-relative directories to be ignored by the builder. This is processed by the Bndtools builder in Eclipse and the Bnd Gradle plugin for workspace model builds.",
 			BUILDERIGNORE + "=${if;${driver;gradle};bin,bin_test,generated;build}", null, null),
 		new Syntax(BUMPPOLICY, "Sets the version bump policy. This is a parameter to the ${version} macro.",
 			BUMPPOLICY + "==+0", "==+,=+0,+00", Pattern.compile("[=+-0][=+-0][=+-0]")),
 
+		new Syntax(BASELINEREPO, "Define the repository to calculate baselining against.", BASELINEREPO + "=Baseline",
+			null, null),
+		new Syntax(BNDDRIVER, "Sets the driver property.", BNDDRIVER + ":" + BNDDRIVER_ECLIPSE,
```
[the remark this answers is not in the document]
Comment (amitjoy) [reply biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java]: But the user can assign a custom value to specify own driver. I had earlier used it for setting the driver to `RemoteCI` to differentiate between local and remote gradle builds.

[the remark this answers is not in the document]
Comment (bjhargrave) [reply biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java]: That is true. I guess we err on the side of documenting it.

File: biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java [hunk 4]
```diff
@@ -203,18 +220,24 @@
...
 		),
+		new Syntax(TESTCASES, "The " + TESTCASES + " instruction is used to specify the list of test cases to run.",
+			TESTCASES + ": com.foo.bar.MyTest", null, Verifier.FQNPATTERN),
 		new Syntax(CONDITIONAL_PACKAGE, "The " + CONDITIONAL_PACKAGE
 			+ " works as private package but will only include the packages when they are imported. When this header is used, bnd will recursively add packages that match the patterns until there are no more additions.",
 			CONDITIONAL_PACKAGE + ": com.*", "${packages}", null),
 		new Syntax(CONDITIONALPACKAGE, "The " + CONDITIONALPACKAGE
 			+ " works as private package but will only include the packages when they are imported. When this header is used, bnd will recursively add packages that match the patterns until there are no more additions.",
 			CONDITIONALPACKAGE + ": com.*", "${packages}", null),
-		new Syntax(PRIVATE_PACKAGE, "The " + PRIVATE_PACKAGE
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java]: This changes seems to have deleted `PRIVATE_PACKAGE` entirely which is incorrect.

Comment (amitjoy) [reply biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java]: PRIVATE_PACKAGE has been duplicated mistakenly and that's why removed the redundant entry.

Comment (bjhargrave) [reply biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java]: If I look at the complete file from your commit, https://github.com/bndtools/bnd/blob/8cb860894495baabda7a0435062497e8a491b8e5/biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java, I do not see `PRIVATE_PACKAGE` anywhere. Also, I only see `PRIVATE_PACKAGE` once in the current master.

Comment (amitjoy) [reply biz.aQute.bndlib/src/aQute/bnd/help/Syntax.java]: ah my bad. fixed.

**Comment by bjhargrave** [comment]
Thanks!

**Pull request merged**